NorthstarDLL/mods/autodownload/moddownloader.cpp

```diff
@@ -0,0 +1,525 @@
+#include "moddownloader.h"
+#include <rapidjson/fwd.h>
+#include <mz_strm_mem.h>
+#include <mz.h>
+#include <mz_strm.h>
+#include <mz_zip.h>
+#include <mz_compat.h>
+#include <thread>
+#include <future>
+#include <bcrypt.h>
+#include <winternl.h>
+#include <fstream>
+
+ModDownloader* g_pModDownloader;
+
+ModDownloader::ModDownloader()
+{
+	spdlog::info("Mod downloaded initialized");
+	modState = {};
+}
+
+size_t writeToString(void* ptr, size_t size, size_t count, void* stream)
+{
+	((std::string*)stream)->append((char*)ptr, 0, size * count);
+	return size * count;
+}
+
+void ModDownloader::FetchModsListFromAPI()
+{
+	std::thread requestThread(
+		[this]()
+		{
+			CURLcode result;
+			CURL* easyhandle;
+			rapidjson::Document verifiedModsJson;
+			std::string url = MODS_LIST_URL;
+
+			curl_global_init(CURL_GLOBAL_ALL);
+			easyhandle = curl_easy_init();
+			std::string readBuffer;
+
+			// Fetching mods list from GitHub repository
+			curl_easy_setopt(easyhandle, CURLOPT_CUSTOMREQUEST, "GET");
+			curl_easy_setopt(easyhandle, CURLOPT_TIMEOUT, 30L);
+			curl_easy_setopt(easyhandle, CURLOPT_URL, url.c_str());
+			curl_easy_setopt(easyhandle, CURLOPT_FAILONERROR, 1L);
+			curl_easy_setopt(easyhandle, CURLOPT_WRITEDATA, &readBuffer);
+			curl_easy_setopt(easyhandle, CURLOPT_WRITEFUNCTION, writeToString);
+			result = curl_easy_perform(easyhandle);
+
+			if (result == CURLcode::CURLE_OK)
+			{
+				spdlog::info("Mods list successfully fetched.");
+			}
+			else
+			{
+				spdlog::error("Fetching mods list failed: {}", curl_easy_strerror(result));
+				goto REQUEST_END_CLEANUP;
+			}
+
+			// Load mods list into local state
+			spdlog::info("Loading mods configuration...");
+			verifiedModsJson.Parse(readBuffer);
+			for (auto i = verifiedModsJson.MemberBegin(); i != verifiedModsJson.MemberEnd(); ++i)
+			{
+				std::string name = i->name.GetString();
+				std::string dependency = i->value["DependencyPrefix"].GetString();
+
+				std::unordered_map<std::string, VerifiedModVersion> modVersions;
+				rapidjson::Value& versions = i->value["Versions"];
+				assert(versions.IsArray());
+				for (rapidjson::Value::ConstValueIterator itr = versions.Begin(); itr != versions.End(); ++itr)
+				{
+					const rapidjson::Value& attribute = *itr;
+					assert(attribute.IsObject());
+					std::string version = attribute["Version"].GetString();
+					std::string checksum = attribute["Checksum"].GetString();
+					modVersions.insert({version, {.checksum = checksum}});
+				}
+
+				VerifiedModDetails modConfig = {.dependencyPrefix = dependency, .versions = modVersions};
+				verifiedMods.insert({name, modConfig});
+				spdlog::info("==> Loaded configuration for mod \"" + name + "\"");
+			}
+
+			spdlog::info("Done loading verified mods list.");
+
+		REQUEST_END_CLEANUP:
+			curl_easy_cleanup(easyhandle);
+		});
+	requestThread.detach();
+}
+
+size_t writeData(void* ptr, size_t size, size_t nmemb, FILE* stream)
```

Comment:
```suggestion
size_t WriteData(void* ptr, size_t size, size_t nmemb, FILE* stream)
```
PascalCase for function names

Reply:
Addressed in 7b38a4b60eb550df630506cdf689b94fab99165c